### PR TITLE
Port android support design module to rxjava 2

### PR DESCRIPTION
**Pull request by marcinsus**
My objections:
Should we replace deprecated methods like setOnTabSelectedListener by addOnTabSelectedListener?
What suggested when view action callback is not interface but abstract class?

### CODE REVIEW

File: rxbinding-design/src/main/java/com/jakewharton/rxbinding2/support/design/widget/AppBarLayoutOffsetChangeObservable.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+import android.support.design.widget.AppBarLayout;
+import io.reactivex.Observable;
+import io.reactivex.Observer;
+import io.reactivex.android.MainThreadDisposable;
+
+import static io.reactivex.android.MainThreadDisposable.verifyMainThread;
+
+final class AppBarLayoutOffsetChangeObservable extends Observable<Integer> {
+  final AppBarLayout view;
```
Comment (OrdonTeam) [inline rxbinding-design/src/main/java/com/jakewharton/rxbinding2/support/design/widget/AppBarLayoutOffsetChangeObservable.java]: Add private

File: rxbinding-design/src/main/java/com/jakewharton/rxbinding2/support/design/widget/BottomNavigationViewItemSelectionsObservable.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import android.view.Menu;
+import android.view.MenuItem;
+import io.reactivex.Observable;
+import io.reactivex.Observer;
+import io.reactivex.android.MainThreadDisposable;
+
+import static io.reactivex.android.MainThreadDisposable.verifyMainThread;
+
+final class BottomNavigationViewItemSelectionsObservable extends Observable<MenuItem> {
+  final BottomNavigationView view;
```
Comment (OrdonTeam) [inline rxbinding-design/src/main/java/com/jakewharton/rxbinding2/support/design/widget/BottomNavigationViewItemSelectionsObservable.java]: Add private

File: rxbinding-design/src/main/java/com/jakewharton/rxbinding2/support/design/widget/NavigationViewItemSelectionsObservable.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import android.view.Menu;
+import android.view.MenuItem;
+import io.reactivex.Observable;
+import io.reactivex.Observer;
+import io.reactivex.android.MainThreadDisposable;
+
+import static io.reactivex.android.MainThreadDisposable.verifyMainThread;
+
+final class NavigationViewItemSelectionsObservable extends Observable<MenuItem> {
+  final NavigationView view;
```
Comment (OrdonTeam) [inline rxbinding-design/src/main/java/com/jakewharton/rxbinding2/support/design/widget/NavigationViewItemSelectionsObservable.java]: Add private

File: rxbinding-design/src/main/java/com/jakewharton/rxbinding2/support/design/widget/SnackbarDismissesObservable.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+import android.support.design.widget.Snackbar;
+import android.support.design.widget.Snackbar.Callback;
+import io.reactivex.Observable;
+import io.reactivex.Observer;
+import io.reactivex.android.MainThreadDisposable;
+
+import static io.reactivex.android.MainThreadDisposable.verifyMainThread;
+
+final class SnackbarDismissesObservable extends Observable<Integer> {
+  final Snackbar view;
```
Comment (OrdonTeam) [inline rxbinding-design/src/main/java/com/jakewharton/rxbinding2/support/design/widget/SnackbarDismissesObservable.java]: Add private

File: rxbinding-design/src/main/java/com/jakewharton/rxbinding2/support/design/widget/SnackbarDismissesObservable.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    verifyMainThread();
+    Listener listener = new Listener(view, observer);
+    observer.onSubscribe(listener);
+    view.setCallback(listener.callback);
+  }
+
+  final class Listener extends MainThreadDisposable {
+    private final Snackbar snackbar;
+    private Observer<? super Integer> observer;
+    private final Callback callback = new Callback() {
```
Comment (OrdonTeam) [inline rxbinding-design/src/main/java/com/jakewharton/rxbinding2/support/design/widget/SnackbarDismissesObservable.java]: Move initialization to constructor. You can remove `observer` field then. If not add final to `observer` field.

File: rxbinding-design/src/main/java/com/jakewharton/rxbinding2/support/design/widget/TabLayoutSelectionEventObservable.java [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+import android.support.design.widget.TabLayout.Tab;
+import io.reactivex.Observable;
+import io.reactivex.Observer;
+import io.reactivex.android.MainThreadDisposable;
+
+import static com.jakewharton.rxbinding2.support.design.widget.TabLayoutSelectionEvent.*;
+import static io.reactivex.android.MainThreadDisposable.verifyMainThread;
+
+final class TabLayoutSelectionEventObservable extends Observable<TabLayoutSelectionEvent> {
+  final TabLayout view;
```
Comment (OrdonTeam) [inline rxbinding-design/src/main/java/com/jakewharton/rxbinding2/support/design/widget/TabLayoutSelectionEventObservable.java]: Add private

File: rxbinding-design/src/main/java/com/jakewharton/rxbinding2/support/design/widget/TabLayoutSelectionsObservable.java [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+import android.support.design.widget.TabLayout;
+import android.support.design.widget.TabLayout.Tab;
+import io.reactivex.Observable;
+import io.reactivex.Observer;
+import io.reactivex.android.MainThreadDisposable;
+
+import static io.reactivex.android.MainThreadDisposable.verifyMainThread;
+
+final class TabLayoutSelectionsObservable extends Observable<Tab> {
+  final TabLayout view;
```
Comment (OrdonTeam) [inline rxbinding-design/src/main/java/com/jakewharton/rxbinding2/support/design/widget/TabLayoutSelectionsObservable.java]: Add private

**Comment by OrdonTeam** [comment]
@ghostbuster91 resolved abstract class problem in this commit 4e951cba314d2cf736d57c2425c58252676020be
We are open for any suggestion on this matter.

**Comment by OrdonTeam** [comment]
"Should we replace deprecated methods like setOnTabSelectedListener by addOnTabSelectedListener?"
I suppose it is not best time for resolving those issues.

**Comment by JakeWharton** [comment]
@OrdonTeam feel free to clean up the commented issues in a separate PR. I want to keep velocity moving forward.

And yes, we should prefer add vs. set when it's in the support libraries.

**Pull request merged**